data/src/main/java/org/apache/iceberg/data/PartitionStatsHandler.java

```diff
@@ -61,30 +61,29 @@ public class PartitionStatsHandler {
 
   private PartitionStatsHandler() {}
 
-  public enum Column {
-    PARTITION(0),
-    SPEC_ID(1),
-    DATA_RECORD_COUNT(2),
-    DATA_FILE_COUNT(3),
-    TOTAL_DATA_FILE_SIZE_IN_BYTES(4),
-    POSITION_DELETE_RECORD_COUNT(5),
-    POSITION_DELETE_FILE_COUNT(6),
-    EQUALITY_DELETE_RECORD_COUNT(7),
-    EQUALITY_DELETE_FILE_COUNT(8),
-    TOTAL_RECORD_COUNT(9),
-    LAST_UPDATED_AT(10),
-    LAST_UPDATED_SNAPSHOT_ID(11);
-
-    private final int id;
-
-    Column(int id) {
-      this.id = id;
-    }
-
-    public int id() {
-      return id;
-    }
-  }
+  public static final int PARTITION_FIELD_ID = 0;
```

Comment:
Why is this starting with 0? In the previous code the id was started with 1

Reply:
`PartitionStats.set(pos, value)` was based on enum ordinal before (starts from zero). 

Now that we don't have ordinal, I have used field_id instead as `field_id-1` may look odd.
I even thought of updating the `PartitionStats.set(pos, value)` to set from 1, but all of the places `Structlike` starts from 0. So, didn't went with that path.

Reply:
Also, since it is an unreleased feature, modifying this won't cause compatibility issues.